src/dev/typescript/get_ts_project_for_absolute_path.ts

```diff
@@ -31,8 +31,9 @@ import { PROJECTS } from './projects';
  *
  * @param path Absolute path to a .ts file
  */
-export function getTsProjectForAbsolutePath(path: string) {
+export function getTsProjectForAbsolutePath(path: string): Project {
   const relPath = relative(REPO_ROOT, path);
+
   const projects = PROJECTS.filter(p => p.isAbsolutePathSelected(path));
```

Comment:
Personally, I'd prefer if we defined this filter function as a type guard so that TS can infer the return value, and so that `projects` has the right type inside the function.

Reply:
I'm not sure how you would use a type guard here since `projects` can only ever be one type (`Projects[]`). TS correctly infers this as being of type `Project` since the `filter` function is typed.

I haven't given this much thought, but I think I prefer exported functions explicitly defining their return types instead of inferring it from returned values only. If you change the implementation of the function, you will immediately see a warning that you've broken the type contract as you're working inside of that file. If the type is inferred, consumers of the function would throw type errors, but you probably won't notice this as quickly.

Reply:
We don't have a defined convention here. I think if that's one we want to adopt, we should introduce a PR with a linter rule for it, allowing for a team discussion. IMO anything that is not enforced by a lint rule is free game. This philosophy encourages us to have one discussion about it, make a decision, and commit to it rather than having the same argument over and over.

Reply:
Yeah, I was wrong in my original comment, and I agree @joshdover.